### PR TITLE
CI workflow for fuzzing

### PR DESCRIPTION
**Pull request by robertwoj-microsoft**
## Description

CI workflow for fuzzing.

*Note:* Reopened https://github.com/Azure/azure-osconfig/pull/792 after rebasing with main. There is no dependency on https://github.com/Azure/azure-osconfig/pull/791 now and these changes can be merged.

## Checklist

- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-osconfig/blob/main/CONTRIBUTING.md).
- [x] All unit tests are passing.
- [x] I have merged the latest `main` branch prior to this PR submission.
- [x] I submitted this PR against the `main` branch.

### CODE REVIEW

File: .github/workflows/fuzzing-run.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+      - uses: actions/download-artifact@v4
+        id: download
+        with:
+          name: ${{ inputs.target }}-fuzzer
+
+      - name: Install prerequisites
+        run: |
+          sudo apt update
+          sudo apt install -y clang
```
Comment (AhmedBM) [inline .github/workflows/fuzzing-run.yml:45]: Unnecessary since we have it in the container now

File: .github/workflows/package-build.yml [hunk 2]
```diff
@@ -98,6 +103,8 @@ jobs:
           cmd: |
             if [ "${{ inputs.machine-config }}" = "true" ]; then
               cmake ../src -DCMAKE_BUILD_TYPE=${{ env.BUILD_TYPE }}
+            elif [ "${{ inputs.build-fuzzers }}" = "true" ]; then
+              cmake ../src -DCMAKE_BUILD_TYPE=Debug -DCMAKE_C_COMPILER="/usr/bin/clang" -DCMAKE_CXX_COMPILER="/usr/bin/clang++" -DTWEAK_VERSION=${{ steps.version.outputs.tweak }} -Duse_prov_client=ON -Dhsm_type_symm_key=ON -DCOMPILE_WITH_STRICTNESS=ON -DBUILD_TESTS=OFF -DBUILD_MODULETEST=ON -DBUILD_SAMPLES=OFF -Duse_default_uuid=ON -DBUILD_ADAPTERS=ON -DBUILD_FUZZER=ON
```
Comment (AhmedBM) [inline .github/workflows/package-build.yml:107]: Can we also enable `BUILD_TESTS`?

**Review by AhmedBM — COMMENTED** [review]
just a few minor comments

File: .github/workflows/package-build.yml (hunk 2 above)
Comment (MariusNi) [reply .github/workflows/package-build.yml:107]: Or, can we also exclude BUILD_MODULETEST (why would we need this one for fuzzing)?

Comment (robertwoj-microsoft) [reply .github/workflows/package-build.yml:107]: Will disable BUILD_MODULETEST.

File: .github/workflows/fuzzing-run.yml (hunk 1 above)
Comment (robertwoj-microsoft) [reply .github/workflows/fuzzing-run.yml:45]: This has been discussed under https://github.com/Azure/azure-osconfig/pull/792 which got closed after rebasing with main. Needed at runtime, let's keep it now.

**Review by MariusNi — APPROVED** [review]
:shipit:

**Pull request merged**